st2common/st2common/models/db/__init__.py

```diff
@@ -444,34 +453,37 @@ def _get_ssl_kwargs(
 ):
     # NOTE: In pymongo 3.9.0 some of the ssl related arguments have been renamed -
     # https://api.mongodb.com/python/current/changelog.html#changes-in-version-3-9-0
-    # Old names still work, but we should eventually update to new argument names.
+    # Old names stop working in pymongo 4, so we need to migrate now:
+    # https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#renamed-uri-options
     ssl_kwargs = {
-        "ssl": ssl,
+        "tls": ssl,
     }
+    # TODO: replace ssl_keyfile and ssl_certfile with tlsCertificateFile per pymongo:
+    #   > Instead of using ssl_certfile and ssl_keyfile to specify the certificate
+    #   > and private key files respectively, use tlsCertificateKeyFile to pass a
+    #   > single file containing both the client certificate and the private key.
+    # The tlsCertificateFile switch will be user-facing as files must be combined.
     if ssl_keyfile:
-        ssl_kwargs["ssl"] = True
+        ssl_kwargs["tls"] = True
         ssl_kwargs["ssl_keyfile"] = ssl_keyfile
     if ssl_certfile:
-        ssl_kwargs["ssl"] = True
+        ssl_kwargs["tls"] = True
         ssl_kwargs["ssl_certfile"] = ssl_certfile
```

Comment:
Upgrade notes should explain the pymongo requirement and instruct the user to `cat client.key client.crt > client_combined.pem` and update `st2.conf` to remove the old key/cert `ssl_keyfile` and `ssl_certfile` entries and replace them with a single entry:

```tlscertificatekeyfile = client_combined.pem```

If the code allows it, we could check if `st2.conf` contains `ssl_keyfile` or `ssl_certfile` and raise an error to prevent StackStorm from starting, but the current behaviour is to simply ignore unknown key-pairs in the configuration file, so it may be non-trivial to implement.

Reply:
It may be easier as well for rolling upgrades to allow them to add in the new parameters and keep the old ones, that way the new ones will only be read by the instances that have upgraded and ignore the others.
So I think we just need to make sure it goes into the upgrade notes as a breaking change

Reply:
Makes sense. Should we add all the other `tls*` variants to st2.conf as well?

Maybe there is a way to mark an option deprecated, which would result in a non-fatal warning...

Reply:
OK. I renamed the ssl vars. Please take another look.